source/jormungandr/jormungandr/realtime_schedule/timeo.py

```diff
@@ -217,9 +218,9 @@ def _make_url(self, route_point, count=None, from_dt=None):
         # timeo can only handle items_per_schedule if it's < 5
         count = min(count or 5, 5)# if no value defined we ask for 5 passages
 
-        # if a custom datetime is provided we give it to timeo
+        # if a custom datetime is provided we give it to timeo but we round it to improve cachability
         dt_param = '&NextStopReferenceTime={dt}'\
-            .format(dt=self._timestamp_to_date(from_dt).strftime('%Y-%m-%dT%H:%M:%S')) \
+            .format(dt=floor_datetime(self._timestamp_to_date(from_dt), self.step).strftime('%Y-%m-%dT%H:%M:%S')) \
```

Comment:
Then we should also filter afterward, removing those in interval `[floor(from_dt), from_dt[` no?

Reply:
We could, but we didn't do it before when we were thinking that the cache was working